### PR TITLE
Move /results & /evals reqs into unsafe methods

### PR DESCRIPTION
**Pull request by nicolewhite**
Same as https://github.com/autoblocksai/python-sdk/pull/135

### CODE REVIEW

File: src/testing/run.ts [hunk 1]
```diff
@@ -102,22 +77,31 @@ async function sendError(args: {
...
       output: args.output,
     });
   });
+
+  await client.post('/evals', {
+    testExternalId: args.testId,
+    testCaseHash: args.testCaseHash,
+    evaluatorExternalId: args.evaluator.id,
+    score: evaluation.score,
+    threshold: evaluation.threshold,
```
Comment (dmorton2297) [inline src/testing/run.ts:103]: We should be posting evaluation metadata as well

File: src/testing/run.ts [hunk 2]
```diff
@@ -143,18 +126,7 @@ async function runEvaluator<TestCaseType, OutputType>(args: {
...
-
-  if (evaluation === undefined) return;
-
-  await client.post('/evals', {
-    testExternalId: args.testId,
-    testCaseHash: args.testCaseHash,
-    evaluatorExternalId: args.evaluator.id,
-    score: evaluation.score,
-    threshold: evaluation.threshold,
-  });
```
Comment (dmorton2297) [inline src/testing/run.ts:157]: fyi - I have this line updated in my PR to include metadata

File: src/testing/util.ts [hunk 3]
```diff
@@ -79,3 +81,27 @@ export class Semaphore {
...
+    | ((testCase: TestCaseType) => string),
+): string {
+  if (Array.isArray(testCaseHash)) {
+    const concatenated = testCaseHash
+      .map((key) => JSON.stringify(testCase[key]))
+      .join('');
+    return crypto.createHash('md5').update(concatenated).digest('hex');
+  } else {
+    return testCaseHash(testCase);
+  }
```
Comment (dmorton2297) [inline src/testing/util.ts:106]: nit - I prefer to drop the else and just return

File: test/testing/run.spec.ts [hunk 4]
```diff
@@ -709,4 +710,70 @@ describe('Testing SDK', () => {
...
+        id: 'my-test-id',
+        testCases: [{ x: 1, y: 2 }],
+        testCaseHash: ['x', 'y'],
+        evaluators: [],
+        fn: ({ testCase }: { testCase: MyTestCase }) => ({
+          result: `${testCase.x} + ${testCase.y} = ${testCase.x + testCase.y}`,
+        }),
+      });
+
+      // console.log('isPrimitiveSpy', isPrimitiveSpy.mock.calls)
```
Comment (dmorton2297) [inline test/testing/run.spec.ts:750]: remove

File: src/testing/run.ts (hunk 2 above)
Comment (nicolewhite) [reply src/testing/run.ts:157]: yep np, i'll resolve when the merge conflicts pops up

**Review by adamnolte — APPROVED** [review]

**Pull request merged**